dump.go

```diff
@@ -137,3 +110,59 @@ func display(o io.Writer, f io.Reader, fname string) error {
 	fmt.Fprintf(o, "data = %v\n", rv.Elem().Interface())
 	return nil
 }
+
+func sizeof(r io.ReaderAt) (int64, error) {
+	switch r := r.(type) {
+	case interface{ Stat() (os.FileInfo, error) }:
+		fi, err := r.Stat()
+		if err != nil {
+			return 0, err
+		}
+		return fi.Size(), nil
+	case io.Seeker:
+		pos, err := r.Seek(0, io.SeekCurrent)
+		if err != nil {
+			return 0, err
+		}
+		sz, err := r.Seek(0, io.SeekEnd)
+		if err != nil {
+			return 0, err
+		}
+		_, err = r.Seek(pos, io.SeekStart)
+		if err != nil {
+			return 0, err
+		}
+		return sz, nil
+	default:
+		return 0, fmt.Errorf("npyio: unsupported  reader: %T", r)
+	}
+}
+
+func UnzipNpz(r io.ReaderAt) ([]*zip.File, error) {
```

Comment:
please document this new exported function, trying to follow Go conventions.
(particularly, documenting that users would need to call close on each of the `zip.File`s)

also, it would be great to re-use within the `Dump` function.